config.yml

```diff
@@ -27,8 +27,11 @@ service:
   job_cleanup_interval: 300
   log_level: INFO
 
-connector_id: 'changeme'
-service_type: 'changeme'
+#connectors:
+#  changeme:
+#    service_type: changeme
+#    elasticsearch:
+#      api_key: changeme
```

Comment:
Question:

Just to make sure I understand, if I have an SPO connector with id `abcd` I can set configuration like so:

```
connectors:
  abcd:
    service_type: 'sharepoint_online'
```

If that's correct, that leaves me with two questions:

- Is the `elasticsearch.api_key` required now?
- Will we have any issue using the connector id as a key in a yaml file, as it's randomised string?

Reply:
>Is the `elasticsearch.api_key` required now?

The top-level `elasticsearch` credential (either `api_key` or `username`/`password` pair) is always needed, which is used to access data from system indices (i.e. `.elastic-connectors` and `.elastic-connectors-sync-jobs`), and the API key generated from Kibana has sufficient privileges.

In the per connectors config, API key is also required, e.g. 
```yaml
connectors:
  foo: # suppose index name is search-foo
    service_type: mysql
    elasticsearch:
      api_key: foo
  bar: # suppose index name is search-bar
    service_type: mongodb
    elasticsearch:
      api_key: bar
```

API key `foo` will have access to index `search-foo` but not `search-bar` and vice versa.

>Will we have any issue using the connector id as a key in a yaml file, as it's a randomised string?

The connector id is an auto-generated id from Elasticsearch. From the [doc](https://www.elastic.co/guide/en/elasticsearch/guide/current/index-doc.html#_autogenerating_ids),:
>Autogenerated IDs are 20 character long, URL-safe, Base64-encoded GUID strings.

This is similar to `sources` field. Do you foresee any possible scenario where a connector id can break the yaml format?

Reply:
> This is similar to sources field. Do you foresee any possible scenario where a connector id can break the yaml format?

Thanks for the explanation and the docs link. I don't foresee any issues now.
That being said, I also agree with the conversation below about expressive yaml being more user-friendly. If we were to choose between the two options, I would prefer not using the id as a key.